sqliterepo/repository.c

```diff
@@ -481,36 +481,49 @@ sqlx_repository_configure_type(sqlx_repository_t *repo,
 }
 
 void
-sqlx_repository_configure_close_callback(sqlx_repository_t *repo,
-		sqlx_repo_close_hook cb, gpointer cb_data)
+sqlx_repository_configure_db_check_opening_callback(
+		sqlx_repository_t *repo,
+		sqlx_repo_db_check_opening_hook cb, gpointer cb_data)
```

Comment:
I would have kept `sqlx_repository_configure_open_callback` and `sqlx_repo_open_hook`, with the addition of the open mode.